api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
I keep thinking about intent vs. actual, like the state model in kubernetes. Context.Spec.Breakpoints vs Context.Status.Breakpoints rather than separate/disassociated objects. Is that going to far?